crates/ol/stf/src/account_processing.rs

```diff
@@ -0,0 +1,140 @@
+        BRIDGE_GATEWAY_ACCT_ID => {
+            handle_bridge_gateway_message(state, sender, msg, context)?;
+        }
+
+        // Any other address we assume is a ledger account, so we have to look it up.
+        _ => {
+            // TODO adapt the account state traits to make this all more
+            // amendable to avoiding copies/clones
+
+            // Make a copy of the account state.  I don't love this.
+            let Some(mut acct_state) = state.get_account_state(target)?.cloned() else {
+                // If we don't find it then we can just ignore it.
+                // TODO do something with the funds we're throwing away by doing this
+                return Ok(());
+            };
+
+            // First, just increase the balance right now.
+            let coin = Coin::new_unchecked(msg.value());
+            acct_state.add_balance(coin);
+
+            // Then depending on the type we call a different handler function
+            // for postprocessing.
+            let mut tystate = acct_state.get_type_state().expect("stf/acct: type state");
+            match &mut tystate {
+                AccountTypeState::Empty => {
+                    // Do nothing.
+                }
+
+                AccountTypeState::Snark(sastate) => {
+                    // Call the handler fn now that we've increased the balance.
+                    handle_snark_account_message(state, sastate, sender, msg, context)?;
+                }
+            }
+
+            // Update with the now-modified type state.
+            acct_state.set_type_state(tystate)?;
+
+            // Then write the whole account back with the changes.
+            state.update_account_state(target, acct_state)?;
+        }
+    }
+
+    Ok(())
+}
+
+fn handle_bridge_gateway_message<S: StateAccessor>(
+    state: &mut S,
+    sender: AccountId,
+    payload: MsgPayload,
+    context: &BasicExecContext<'_>,
+) -> ExecResult<()> {
+    // Parse the message from the payload data.
+    let Ok(msg) = MsgRef::try_from(payload.data()) else {
+        // Invalid message format, just ignore.
+        return Ok(());
+    };
+
+    if msg.ty() != WITHDRAWAL_MSG_TYPE_ID {
+        // Some other message type, just ignore.
+        return Ok(());
+    }
+
+    let Some(withdrawal_data) = msg.try_as_withdrawal() else {
+        // Invalid withdrawal message, just ignore.
+        return Ok(());
+    };
+
+    // Check if the withdrawal amount is in allowed denominations
+    let withdrawal_amt = payload.value();
+
+    // TODO move to params struct
+    let withdrawal_denoms = &[100_000_000];
+
+    // Make that the amount is an appropriate denomination.
+    if !withdrawal_denoms.contains(&withdrawal_amt.into()) {
+        return Ok(());
+    }
+
+    // If it is, then we can emit a OL log with the amount and destination.
+    let log_data = SimpleWithdrawalIntentLogData {
+        amt: withdrawal_amt.into(),
+        dest: withdrawal_data.dest_desc.clone(),
```

Comment:
shouldn't we also check correctness of destination address (I think that now we only check address size)?